mockito-core/build.gradle.kts

```diff
@@ -200,3 +204,47 @@ fun mockitoExtensionConfigFile(project: Project, mockitoExtension: String) =
     file(project.layout.buildDirectory.dir("generated/resources/ext-config/test/mockito-extensions/$mockitoExtension"))
 //</editor-fold>
 
+//<editor-fold defaultstate="collapsed" desc="Generate version class file">
+// Manifest cannot be used because Android strips the manifest from the jar.
+
+val generatedVersionClassDir: Provider<Directory> = layout.buildDirectory.dir("generated/sources/version/java")
+
+sourceSets.main {
+    java {
+        srcDir(generatedVersionClassDir)
+    }
+}
+
+tasks {
+    val generateVersionClass by registering {
+        val versionClassContent = """
+        package org.mockito.internal.util;
+        class MockitoVersion {
+            public static final String VERSION = "${project.version}";
+        }
+        """.trimIndent()
+
+        val versionClass = file(generatedVersionClassDir.map { it.file("org/mockito/internal/util/MockitoVersion.java") })
+
+        doLast {
+            versionClass.run {
+                parentFile.mkdirs()
+                createNewFile()
+                writeText(versionClassContent)
+            }
+        }
+    }
+    classes.dependsOn(generateVersionClass)
+
+    rootProject.idea {
```

Comment:
Rather than manually configuring the dependency here, I think we can use the output of the Task declared here as inputs of the `srcDir`: https://discuss.gradle.org/t/right-way-to-generate-sources-in-gradle-7/41141/3

E.g. I think the folowing code is equivalent to what you wrote:

```kotlin

val generatedVersionClassDir: Provider<Directory> = layout.buildDirectory.dir("generated/sources/version/java")

tasks {
    val generateVersionClassTask by registering {
        val versionClassContent = """
        package org.mockito.internal.util;
        class MockitoVersion {
            public static final String VERSION = "${project.version}";
        }
        """.trimIndent()

        val versionClass = file(generatedVersionClassDir.map { it.file("org/mockito/internal/util/MockitoVersion.java") })

        doLast {
            versionClass.run {
                parentFile.mkdirs()
                createNewFile()
                writeText(versionClassContent)
            }
        }
    }

    sourceSets.main {
        java {
            srcDir(generateVersionClassTask)
        }
    }
}
```

This also avoids us adding the IntelliJ extension and needing to use `afterSync`, which afaik is a performance penalty.

Also do note that this only works if the task has a configured `@OutputDirectory`. Therefore, we most likely need to create a custom Gradle task in `buildSrc/src/main/kotlin` and then configure the task here. Something like this:

```kotlin
tasks {
    // I don't know what the correct Kotlin way is to declare this task, as I only used Groovy Gradle
    val generateVersionClassTask by registering (GenerateMockitoVersionClass) {`
        mockitoVersion = project.version
        generatedVersionClassDir = layout.buildDirectory.dir("generated/sources/version/java")
    }

    sourceSets.main {
        java {
            srcDir(generateVersionClassTask)
        }
    }
}
```

Here `GenerateMockitoVersionClass` has an `@OutputDirectory` of `generatedVersionClassDir` and an `@Input` for `mockitoVersion`

Reply:
Interesting I'll take a look

Reply:
Done